csxtools/utils.py

```diff
@@ -76,20 +77,28 @@ def get_fastccd_images(light_header, dark_headers=None,
 
         bgnd = np.array(dark)
 
-    data = _get_images(light_header, tag)
-    data = correct_images(data, bgnd, gain=gain)
+    data, n_images = _get_images(light_header, tag)
+    data = correct_images(data, bgnd, flat=flat, gain=gain)
     data = rotate90(data, 'cw')
-    return data
+    return data, n_images
```

Comment:
@vivekthampy, sorry my question was why return something that is completely derived from the data. Can't you just get n_images from the data by doing 

```python
n_images = [len(i) for i in images]
```

Or am I missing something. I don't like the idea of returning something that is just derived from the other variable you are passing. 

I think I need to understand the code better, it is not obvious to me. I understand what you want to do and why.

You should also put comments here because the code is not obvious at all.